jax/_src/nn/functions.py

```diff
@@ -431,17 +447,17 @@ def softmax(x: Array,
     :func:`log_softmax`
   """
   if jax.config.jax_softmax_custom_jvp:
-    return _softmax(x, axis, where, initial)
+    return _softmax(x, axis, where, initial)  # type: ignore[return-value]
```

Comment:
Do you understand why this ignore statement is necessary? It looks like `_softmax` is annotated to return `Array`

Reply:
Not entirely, but it seem mypy is confused by the `@functools.partial` decorator in `_softmax`

```
jax/_src/nn/functions.py:450: error: Incompatible return value type (got "ReturnValue", expected "Array")  [return-value]
```

`ReturnValue` in the error message is a type parameter of `jax.custom_jvp`, which mypy just keeps as is instead of substituting for the return type of `_softmax` (or for `Any`):

https://github.com/google/jax/blob/2d068a1caa97ebde662604bb95298cff2abb0afa/jax/_src/custom_derivatives.py#L234

Reply:
OK, fair enough. The inner workings of mypy are sometimes mysterious...